src/frontends/tensorflow_common/src/op/scatter_nd.cpp

```diff
@@ -15,10 +17,30 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 OutputVector translate_scatter_nd_op(const NodeContext& node) {
-    default_op_checks(node, 3, {"ScatterNd", "SCATTER_ND"});
+    default_op_checks(node, 3, {"ScatterNd", "SCATTER_ND"}, true);
     auto input_indices = node.get_input(0);
     auto updates = node.get_input(1);
     auto shape = node.get_input(2);
+    auto complex_type_mark_updates = as_type_ptr<ComplexTypeMark>(updates.get_node_shared_ptr());
+
+    if (complex_type_mark_updates) {
+        updates = complex_type_mark_updates->input_value(0);
+        // Add two auxiliary dimensions to the shape tensor
+        Shape shape_dims = shape.get_shape();
+        shape_dims.insert(shape_dims.begin(), 2);
+        auto aux_shape = create_same_type_const<int32_t>(shape, std::vector<int32_t>{2}, Shape{1});
+        auto updated_shape = make_shared<v0::Concat>(OutputVector{aux_shape, shape}, 0);
```

Comment:
please construct a graph using `ShapeOf` and `Concat` operations to append auxiliary dimension